src/main/kotlin/no/nav/syfo/infrastructure/pdl/PdlClient.kt

```diff
@@ -0,0 +1,188 @@
+import no.nav.syfo.api.ClientEnvironment
+import no.nav.syfo.api.httpClientDefault
+import no.nav.syfo.domain.PersonIdent
+import no.nav.syfo.infrastructure.NAV_CALL_ID_HEADER
+import no.nav.syfo.infrastructure.azuread.AzureAdClient
+import no.nav.syfo.infrastructure.azuread.AzureAdToken
+import no.nav.syfo.infrastructure.bearerHeader
+import no.nav.syfo.infrastructure.metric.METRICS_NS
+import no.nav.syfo.infrastructure.metric.METRICS_REGISTRY
+import no.nav.syfo.infrastructure.pdl.dto.*
+import no.nav.syfo.infrastructure.redis.RedisStore
+import org.slf4j.LoggerFactory
+
+class PdlClient(
+    private val azureAdClient: AzureAdClient,
+    private val pdlEnvironment: ClientEnvironment,
+    private val cache: RedisStore,
+) {
+
+    suspend fun getPdlIdenter(
+        personIdent: PersonIdent,
+        callId: String? = null,
+    ): PdlHentIdenter? {
+        val token = azureAdClient.getSystemToken(pdlEnvironment.clientId)
+            ?: throw RuntimeException("Failed to send PdlHentIdenterRequest to PDL: No token was found")
+
+        val query = getPdlQuery(
+            queryFilePath = "/pdl/hentIdenter.graphql",
+        )
+
+        val request = PdlHentIdenterRequest(
+            query = query,
+            variables = PdlHentIdenterRequestVariables(
+                ident = personIdent.value,
+                historikk = true,
+                grupper = listOf(
+                    IdentType.FOLKEREGISTERIDENT,
+                ),
+            ),
+        )
+
+        val response: HttpResponse = httpClient.post(pdlEnvironment.baseUrl) {
+            header(HttpHeaders.Authorization, bearerHeader(token.accessToken))
+            header(HttpHeaders.ContentType, ContentType.Application.Json.toString())
+            header(TEMA_HEADER, ALLE_TEMA_HEADERVERDI)
+            header(NAV_CALL_ID_HEADER, callId)
+            header(IDENTER_HEADER, IDENTER_HEADER)
+            setBody(request)
+        }
+
+        when (response.status) {
+            HttpStatusCode.OK -> {
+                val pdlIdenterResponse = response.body<PdlIdenterDTO>()
+                return if (!pdlIdenterResponse.errors.isNullOrEmpty()) {
+                    Metrics.COUNT_CALL_PDL_IDENTER_FAIL.increment()
+                    pdlIdenterResponse.errors.forEach {
+                        logger.error("Error while requesting IdentList from PersonDataLosningen: ${it.errorMessage()}")
+                    }
+                    null
+                } else {
+                    Metrics.COUNT_CALL_PDL_IDENTER_SUCCESS.increment()
+                    pdlIdenterResponse.data
+                }
+            }
+
+            else -> {
+                Metrics.COUNT_CALL_PDL_IDENTER_FAIL.increment()
+                logger.error("Request to get IdentList with url: ${pdlEnvironment.baseUrl} failed with reponse code ${response.status.value}")
+                return null
+            }
+        }
+    }
+
+    suspend fun navn(personIdent: PersonIdent): String {
```

Comment:
Vet at vi har kalt den `navn` og `person` i andre apper, men synes funksjoner burde være handling/verb

Reply:
Her kan jeg være enig siden det er et `GET` kall. 

Hvis funksjonen derimot ikke egentlig utfører en handling, men for eksempel aksesserer en property synes jeg at selv om det er en funksjon så kan det godt være et substantiv 😊
Eksempel:
```kotlin
class Person(firstName: String, lastName: String) {
	
	// Her hadde jeg tenkt at det blir riktig med substantiv `fullName` vs. `getFullName`
	fun fullName() {
        …
	}
}
```

Reply:
Hmm, jo det kunne man nok.😅 Syns bare å huske at vi har noen `get…` som faktisk ikke er database uthenting eller http requester 🤔
Får ta de eksemplene når de dukker opp. Her er jeg forsåvidt helt enig i at `getNavn` er et bra funksjonsnavn 😊